Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp

```diff
@@ -233,12 +239,14 @@ void SourceBufferPrivateGStreamer::didReceiveSample(Ref<MediaSample>&& sample)
 
 void SourceBufferPrivateGStreamer::didReceiveAllPendingSamples()
 {
-    SourceBufferPrivate::appendCompleted(true);
+    m_appendPromise->resolve();
+    m_appendPromise.reset();
 }
 
 void SourceBufferPrivateGStreamer::appendParsingFailed()
 {
-    SourceBufferPrivate::appendCompleted(false);
+    m_appendPromise->reject(1);
```

Comment:
Should this be `-1`? (Or it doesn't matter because this will be a `PlatformMediaError` in the future.)